libs/vkd3d/state.c

```diff
@@ -812,8 +830,8 @@ static HRESULT d3d12_root_signature_init(struct d3d12_root_signature *root_signa
     /* XXX: Vulkan buffer and image descriptors have different types. In order
      * to preserve compatibility between Vulkan resource bindings for the same
      * root signature, we create descriptor set layouts with two bindings for
-     * each SRV and UAV. */
-    info.descriptor_count += info.srv_count + info.uav_count;
+     * each SRV and UAV. For UAVs we additionally have one UAV counter. */
+    info.descriptor_count += info.srv_count + 2 * info.uav_count;
```

Comment:
Does this also affect root UAVs (which do not have counters?)

Reply:
`srv_count` and `uav_count` only count descriptor table entries, not root descriptors. Those are being counted separately.